packages/babel-core/src/transformation/file/file.js

```diff
@@ -27,10 +27,18 @@ export default class File {
   ast: Object = {};
   scope: Scope;
   metadata: {} = {};
-  hub: Hub = new Hub(this);
   code: string = "";
   inputMap: Object | null = null;
 
+  hub: HubInterface = {
+    // keep it for the usage in babel-core, ex: path.hub.file.opts.filename
+    file: this,
```

Comment:
@loganfsmyth In the other PR you asked to keep `file` here, but I think that it can be removed? (people can already get it in visitors using `this.file`)

Reply:
@nicolo-ribaudo Yeah, my concern was avoiding breaking changes. I think my preference would still be to keep it at this point since we're already pretty close to release. I don't think we'd want it as part of the interface, but leaving it in the object is so easy. Can we leave it with a big comment saying that we should remove it in 8.x but it's being left for backward-compat with 6.x during 7.x's lifetime?

Reply:
Yes, let's avoid breaking things at this point.